src/homepageExperience/components/steps/Finish.tsx

```diff
@@ -24,50 +25,81 @@ export const Finish = (props: OwnProps) => {
   useEffect(() => {
     event(`firstMile.${props.wizardEventName}.finished`)
   }, [])
+
+  const fireConfetti = () => {
```

Comment:
I think we should do two things here:

First, move this function definition outside the component so it's not getting recreated every time this component renders - nothing depends on props, or state, or anything in a component so it can be safely moved outside the render method.

Second, let's create a boolean variable outside the function component with `let` that tells us whether the confetti has been fired yet. Then before calling `fireConfetti`, check to make sure that variable is false. If it's false, fire the confetti, then set the flag to true. This way, if the user goes back and forth between the finish page and another page, they don't have a confetti shower raining down on them every time, which might get annoying. What do you think?

Reply:
I changed the first 👍

I'll slack about the second suggestion, I have an idea that i think might work. Adding a boolean to the file isn't seem to be working because react probably unmounts the component when we switch back and forth.